src/main/kotlin/com/github/imflog/schema/registry/parser/AvroSchemaParser.kt

```diff
@@ -27,20 +27,17 @@ class AvroSchemaParser(
         schemaContent: String,
         localReferences: List<LocalReference>
     ): String {
-        val jsonObj = JSONObject(schemaContent)
-        if (jsonObj.has(FIELDS)) {
-            jsonObj.put(FIELDS,
-                JSONArray(jsonObj
-                    .getJSONArray(FIELDS)
-                    .map { if (it is JSONObject) replaceLocalReference(it, localReferences) else it }
-                )
-            )
-        }
-        return jsonObj.toString()
+        return replaceLocalReference(JSONObject(schemaContent),localReferences).toString()
     }
 
     private fun replaceLocalReference(jsonObject: JSONObject, localReferences: List<LocalReference>): JSONObject {
         when {
+            jsonObject.has(FIELDS) -> jsonObject.put(FIELDS,
+                JSONArray(jsonObject
+                    .getJSONArray(FIELDS)
+                    .map { if (it is JSONObject) replaceLocalReference(it, localReferences) else it }
+                )
```

Comment:
Can't we call the `replaceType` like we do for other types ? It seems that the JSONArray case is already handled is the other method so It could work.

Also, looking at the when, we could probably factorize it (we always do the same thing), but that would just be an optimization :)

Reply:
Hi @ImFlog 

Just push a new version. I have been able to refactor the all method, to only have a `replaceType` method.
I just needed to create a list of the key we want to look at for replacement of localSchema. I didn't put it at first, and I had a infinite loop due to the `name` key that contain the same value than the `type` key.

Waiting on your though on this new version :smiley: !